firefox-ios/Client/Frontend/Library/Bookmarks/Edit Folder/EditFolderViewModel.swift

```diff
@@ -89,7 +96,16 @@ class EditFolderViewModel {
         guard let folder else { return }
         let selectedFolderGUID = selectedFolder?.guid ?? parentFolder.guid
         Task { @MainActor in
-            _ = await bookmarkSaver.save(bookmark: folder, parentFolderGUID: selectedFolderGUID)
+            let result = await bookmarkSaver.save(bookmark: folder, parentFolderGUID: selectedFolderGUID)
+            switch result {
+            case .success(let guid):
+                if let guid = guid {
```

Comment:
Same here i'd use a guard.
Also we can avoid the conditional unwrap now with the equal and just do it like this.
```swift 
guard let guid else { return }
profile.prefs.setString(guid, forKey: ...)
```